### PR TITLE
[TG-2684] Bugfix/fix bootstrapmethods empty optional

### PR DESCRIPTION
**Pull request by mgudemann**
Empty optional return value from parse_lambda_method_handle was treated in a wrong way.

### CODE REVIEW

File: regression/cbmc-java/lambda2/test_no_crash.desc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+CORE
+SymStream.class
+--verbosity 10 --show-goto-functions
```
Comment (thk123) [inline regression/cbmc-java/lambda2/test_no_crash.desc:3]: Perhaps add `EXIT=0` to ensure it really doesn't crash!

**Review by thk123 — APPROVED** [review]
Providing we add the `EXIT=0` this seems like a reasonable fix for the immediate crash problem

File: regression/cbmc-java/lambda2/test_no_crash.desc (hunk 1 above)
Comment (tautschnig) [reply regression/cbmc-java/lambda2/test_no_crash.desc:3]: +1, and add `SIGNAL` - cf. #1905.

**Review by majakusber — APPROVED** [review]
Looks good, I'm also for adding `EXIT=0, SIGNAL=0` for the test that should not be crashing,

File: regression/cbmc-java/lambda2/test_no_crash.desc (hunk 1 above)
Comment (mgudemann) [reply regression/cbmc-java/lambda2/test_no_crash.desc:3]: I completed the regression tests with tests on `EXIT` and `SIGNAL`

**Pull request merged**